### ISSUE TITLE
mapPlot() can report lon -180 where 0 is more sensible

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2156]
Below is an example.   If you change the lower latitude limit to 70, then it's OK.

One kludge to fix this might be just to check if any of the labels is -180, and then to replace it with 0 if doing so makes the value closer to the average of values for adjacent labels.  This would fail if the longitude had been set to the 0 to 360 span, but oce works on the decision that it is on a -180 to 180 span.

``` r
library(oce)
#> Loading required package: gsw
data(coastlineWorld)
par(mar=c(2, 2, 1, 1))
mapPlot(coastlineWorld,
    projection="+proj=stere +lat_ts=80 +lat_N=90",
    latitudelim=c(80, 90), longitudelim=c(-180, 180)) # bad
```

![](https://i.imgur.com/n1OjxZh.png)

``` r
    #latitudelim=c(70, 90), longitudelim=c(-180, 180)) # OK
```

<sup>Created on 2023-09-28 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by dankelley** [comment]
The `debug=1` output includes as below.    It wants to put three nearly-equivalent values at nearly-equivalent zero positions.  The first one wins.  I think the `at` unit is metres, and with a span like this, it would make sense to round to 1 metre, in which case we would have three choices for a label with with identical `at` values.  Then if the choices for label include "0" we could prefer that one.

But I need to think about whether this method would fail for other cases.  I don't plan to work on this until the weekend.

```
    latlabels: TRUE
    lonlabels: TRUE
    axisLabels is not null
    About to consider drawing axis on side 1; next is axisLabels1:
   side value      type            at
2     1  -180 longitude -4.111982e-05
3     1   -45 longitude -1.871073e+06
4     1   -30 longitude -1.130594e+06
5     1   -15 longitude -5.351291e+05
6     1     0 longitude  0.000000e+00
7     1    15 longitude  5.351291e+05
8     1    30 longitude  1.130594e+06
9     1    45 longitude  1.871073e+06
10    1   180 longitude  4.111982e-05
    next is raw skip for side 1:FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
    next is NA-cleaned skip for side 1:FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
```

**Comment by dankelley** [comment]
This has been addressed in "develop" commit 
31ed21c96b89d48d061308048b8541adde0716c6 (with amendment to fix a bug in commit 2606731a5728d592f9427801e66a6c4e344f9eb9).

I think the code should be self-explanatory, but the method is
1. see if any longitude axis value is 0 (prime meridian)
2. if so, see if any other value is either -180 or 180 (dateline, expressed in either way)
3. if so, remove the -180 and/or 180 values from the data frame that gets used to label the axis.

As a test, the code
```R
library(oce)
data(coastlineWorld)
par(mar=c(2, 2, 1, 1))
mapPlot(coastlineWorld,
    projection="+proj=stere +lat_ts=80 +lat_N=90",
    latitudelim=c(80, 90), longitudelim=c(-180, 180), debug=2)
```
produces as below (other information deleted for clarity)
```
      address prime-meridian vs dateline labelling (GH issue 2156)
        longitude=0, index0=5, at0=0
        atSpan=3742147 (in metres, I think)
        criterion0=1e-04
        overlapping=TRUE FALSE FALSE FALSE TRUE FALSE FALSE FALSE TRUE
        valueNumeric=180 45 30 15 0 15 30 45 180
        atDateline=TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE TRUE
        delete=TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE TRUE
      originally, axisLabels1 is
   side value      type            at valueNumeric overlapping atDateline delete
2     1  -180 longitude -4.111982e-05          180        TRUE       TRUE   TRUE
3     1   -45 longitude -1.871073e+06           45       FALSE      FALSE  FALSE
4     1   -30 longitude -1.130594e+06           30       FALSE      FALSE  FALSE
5     1   -15 longitude -5.351291e+05           15       FALSE      FALSE  FALSE
6     1     0 longitude  0.000000e+00            0        TRUE      FALSE  FALSE
7     1    15 longitude  5.351291e+05           15       FALSE      FALSE  FALSE
8     1    30 longitude  1.130594e+06           30       FALSE      FALSE  FALSE
9     1    45 longitude  1.871073e+06           45       FALSE      FALSE  FALSE
10    1   180 longitude  4.111982e-05          180        TRUE       TRUE   TRUE
      after -180/0/180 consideration (github issue 2156), axisLabels1 is
  side value      type         at valueNumeric overlapping atDateline delete
3    1   -45 longitude -1871073.4           45       FALSE      FALSE  FALSE
4    1   -30 longitude -1130593.8           30       FALSE      FALSE  FALSE
5    1   -15 longitude  -535129.1           15       FALSE      FALSE  FALSE
6    1     0 longitude        0.0            0        TRUE      FALSE  FALSE
7    1    15 longitude   535129.1           15       FALSE      FALSE  FALSE
8    1    30 longitude  1130593.8           30       FALSE      FALSE  FALSE
9    1    45 longitude  1871073.4           45       FALSE      FALSE  FALSE
```
Note the 3 columns at the right of the data frame, before/after this consideration of longitudes.

Given the above, and also given that I am not trying for generality in my approach, I judge this issue to have been solved. 

I will, of course, keep an eye out for other problems in maps.
